src/xtgeo/grid3d/grid.py

```diff
@@ -1648,6 +1648,29 @@ def get_heights_above_ffl(
         """
         return _grid_etc1.get_heights_above_ffl(self, ffl=ffl, option=option)
 
+    def get_property_between_surfaces(
+        self,
+        top: xtgeo.RegularSurface,
+        base: xtgeo.RegularSurface,
+        value: int | float = 1,
+        name: str = "between_surfaces",
+    ) -> GridProperty:
+        """Returns a 3D GridProperty object with <value> between two surfaces."
+
+        Args:
+            top: The bounding top surface (RegularSurface object)
+            base: The bounding base surface (RegularSurface object)
+
+        Returns:
+            xtgeo GridProperty object with <value> if cell center is between surfaces,
+            otherwise 0. Note that the property wil be discrete if input value is an
+            integer, otherwise it will be continuous.
```

Comment:
I think we should limit this to only return discrete properties, to make it more consistent for the users 🙂

Reply:
At least I can see myself setting codenames for these properties, and it would be nice if I can rely on the codes property to always be there 👍 
Also RMS does the same by only allowing integers to be set as values, and returning discrete properties from this function.